### ISSUE TITLE
Veyon displays "Debian-gdm" instead of user's session

### ISSUE THREAD

**Issue by brunovila** [veyon/veyon#670]
I have 30 computers in a classroom with Debian 10 installed, and public key access

In some of them I can display the user's desktop normally, but in others I only can see the "Debian gdm" login screen

I tried to restart service manually via SSH

In the syslog I can see:

Computer that works perfectly

```
Oct 14 11:06:16 a13eql19 systemd[1]: Started Veyon Service.
Oct 14 11:06:16 a13eql19 veyon-service[10245]: 2020-10-14T11:06:16.321: [INFO] LinuxServiceCore: Starting server for new session "/org/freedesktop/login1/session/c1" with ID 1 at seat "/org/freedesktop/login1/seat/seat0"
Oct 14 11:06:16 a13eql19 veyon-service[10245]: 2020-10-14T11:06:16.328: [INFO] LinuxServiceCore: stopping server for removed session "/org/freedesktop/login1/session/c1"
Oct 14 11:06:20 a13eql19 veyon-service[10245]: 2020-10-14T11:06:20.333: [WARN] LinuxServiceCore: server for session "/org/freedesktop/login1/session/c1" still running - killing now
Oct 14 11:06:30 a13eql19 veyon-service[10245]: 2020-10-14T11:06:30.343: [WARN] QProcess: Destroyed while process ("/usr/bin/veyon-server") is still running.
Oct 14 11:06:30 a13eql19 veyon-service[10245]: 2020-10-14T11:06:30.349: [INFO] LinuxServiceCore: Starting server for new session "/org/freedesktop/login1/session/_32" with ID 2 at seat "/org/freedesktop/login1/seat/seat0"
```

Computer that does not work

```
Oct 14 11:20:14 a13eql11 systemd[1]: Started Veyon Service.
Oct 14 11:20:14 a13eql11 veyon-service[9315]: 2020-10-14T11:20:14.192: [INFO] LinuxServiceCore: Starting server for new session "/org/freedesktop/login1/session/_32" with ID 2 at seat "/org/freedesktop/login1/seat/seat0"
Oct 14 11:20:14 a13eql11 veyon-service[9315]: 2020-10-14T11:20:14.198: [INFO] LinuxServiceCore: stopping server for removed session "/org/freedesktop/login1/session/_32"
Oct 14 11:20:18 a13eql11 veyon-service[9315]: 2020-10-14T11:20:18.203: [WARN] LinuxServiceCore: server for session "/org/freedesktop/login1/session/_32" still running - killing now
Oct 14 11:20:28 a13eql11 veyon-service[9315]: 2020-10-14T11:20:28.218: [WARN] QProcess: Destroyed while process ("/usr/bin/veyon-server") is still running.
Oct 14 11:20:28 a13eql11 veyon-service[9315]: 2020-10-14T11:20:28.222: [INFO] LinuxServiceCore: Starting server for new session "/org/freedesktop/login1/session/c1" with ID 1 at seat "/org/freedesktop/login1/seat/seat0"
```

VeyonService.log
```
2020-10-14T12:11:59.073: [DEBUG] PluginManager::loadPlugins(): discovered plugin "AuthKeys" at "/usr/lib/x86_64-linux-gnu/veyon/authkeys.so"
2020-10-14T12:11:59.074: [DEBUG] PluginManager::loadPlugins(): discovered plugin "BuiltinX11VncServer" at "/usr/lib/x86_64-linux-gnu/veyon/builtin-x11vnc-server.so"
2020-10-14T12:11:59.074: [DEBUG] PluginManager::loadPlugins(): discovered plugin "BuiltinDirectory" at "/usr/lib/x86_64-linux-gnu/veyon/builtindirectory.so"
2020-10-14T12:11:59.074: [DEBUG] PluginManager::loadPlugins(): discovered plugin "Config" at "/usr/lib/x86_64-linux-gnu/veyon/config.so"
2020-10-14T12:11:59.075: [DEBUG] PluginManager::loadPlugins(): discovered plugin "Demonstration" at "/usr/lib/x86_64-linux-gnu/veyon/demo.so"
2020-10-14T12:11:59.075: [DEBUG] PluginManager::loadPlugins(): discovered plugin "DesktopServices" at "/usr/lib/x86_64-linux-gnu/veyon/desktopservices.so"
2020-10-14T12:11:59.075: [DEBUG] PluginManager::loadPlugins(): discovered plugin "ExternalVncServer" at "/usr/lib/x86_64-linux-gnu/veyon/external-vnc-server.so"
2020-10-14T12:11:59.075: [DEBUG] PluginManager::loadPlugins(): discovered plugin "FileTransfer" at "/usr/lib/x86_64-linux-gnu/veyon/filetransfer.so"
2020-10-14T12:11:59.079: [DEBUG] PluginManager::loadPlugins(): discovered plugin "LDAP Basic" at "/usr/lib/x86_64-linux-gnu/veyon/ldap.so"
2020-10-14T12:11:59.079: [DEBUG] PluginManager::loadPlugins(): discovered plugin "PowerControl" at "/usr/lib/x86_64-linux-gnu/veyon/powercontrol.so"
2020-10-14T12:11:59.079: [DEBUG] PluginManager::loadPlugins(): discovered plugin "RemoteAccess" at "/usr/lib/x86_64-linux-gnu/veyon/remoteaccess.so"
2020-10-14T12:11:59.079: [DEBUG] PluginManager::loadPlugins(): discovered plugin "ScreenLock" at "/usr/lib/x86_64-linux-gnu/veyon/screenlock.so"
2020-10-14T12:11:59.080: [DEBUG] PluginManager::loadPlugins(): discovered plugin "Screenshot" at "/usr/lib/x86_64-linux-gnu/veyon/screenshot.so"
2020-10-14T12:11:59.080: [DEBUG] PluginManager::loadPlugins(): discovered plugin "ServiceControl" at "/usr/lib/x86_64-linux-gnu/veyon/servicecontrol.so"
2020-10-14T12:11:59.080: [DEBUG] PluginManager::loadPlugins(): discovered plugin "Shell" at "/usr/lib/x86_64-linux-gnu/veyon/shell.so"
2020-10-14T12:11:59.081: [DEBUG] PluginManager::loadPlugins(): discovered plugin "SystemUserGroups" at "/usr/lib/x86_64-linux-gnu/veyon/systemusergroups.so"
2020-10-14T12:11:59.081: [DEBUG] PluginManager::loadPlugins(): discovered plugin "TextMessage" at "/usr/lib/x86_64-linux-gnu/veyon/textmessage.so"
2020-10-14T12:11:59.081: [DEBUG] PluginManager::loadPlugins(): discovered plugin "UserSessionControl" at "/usr/lib/x86_64-linux-gnu/veyon/usersessioncontrol.so"
2020-10-14T12:11:59.081: [DEBUG] VeyonCore::initSystemInfo(): "4.4.2" "a13eql11.sanclemente.local" "linux" "5.7.0-0.bpo.2-amd64" "Debian GNU/Linux 10 (buster)" "debian" "10"
2020-10-14T12:11:59.084: [DEBUG] ServiceDataManager::ServiceDataManager():
2020-10-14T12:11:59.085: [DEBUG] LinuxServiceCore::connectToLoginManager(): connected to login manager
2020-10-14T12:11:59.094: [INFO] LinuxServiceCore::startServer(): Starting server for new session "/org/freedesktop/login1/session/_32" with ID 2 at seat "/org/freedesktop/login1/seat/seat0"
2020-10-14T12:11:59.100: [INFO] LinuxServiceCore::stopServer(): stopping server for removed session "/org/freedesktop/login1/session/_32"
2020-10-14T12:12:03.105: [WARN] LinuxServiceCore::stopServer(): server for session "/org/freedesktop/login1/session/_32" still running - killing now
2020-10-14T12:12:13.118: [WARN] QProcess: Destroyed while process ("/usr/bin/veyon-server") is still running.
2020-10-14T12:12:13.121: [INFO] LinuxServiceCore::startServer(): Starting server for new session "/org/freedesktop/login1/session/c1" with ID 1 at seat "/org/freedesktop/login1/seat/seat0"

```

VeyonServer.log
```
2020-10-14T12:12:13.139: [DEBUG] PluginManager::loadPlugins(): discovered plugin "AuthKeys" at "/usr/lib/x86_64-linux-gnu/veyon/authkeys.so"
2020-10-14T12:12:13.140: [DEBUG] PluginManager::loadPlugins(): discovered plugin "BuiltinX11VncServer" at "/usr/lib/x86_64-linux-gnu/veyon/builtin-x11vnc-server.so"
2020-10-14T12:12:13.140: [DEBUG] PluginManager::loadPlugins(): discovered plugin "BuiltinDirectory" at "/usr/lib/x86_64-linux-gnu/veyon/builtindirectory.so"
2020-10-14T12:12:13.140: [DEBUG] PluginManager::loadPlugins(): discovered plugin "Config" at "/usr/lib/x86_64-linux-gnu/veyon/config.so"
2020-10-14T12:12:13.140: [DEBUG] PluginManager::loadPlugins(): discovered plugin "Demonstration" at "/usr/lib/x86_64-linux-gnu/veyon/demo.so"
2020-10-14T12:12:13.140: [DEBUG] PluginManager::loadPlugins(): discovered plugin "DesktopServices" at "/usr/lib/x86_64-linux-gnu/veyon/desktopservices.so"
2020-10-14T12:12:13.141: [DEBUG] PluginManager::loadPlugins(): discovered plugin "ExternalVncServer" at "/usr/lib/x86_64-linux-gnu/veyon/external-vnc-server.so"
2020-10-14T12:12:13.141: [DEBUG] PluginManager::loadPlugins(): discovered plugin "FileTransfer" at "/usr/lib/x86_64-linux-gnu/veyon/filetransfer.so"
2020-10-14T12:12:13.144: [DEBUG] PluginManager::loadPlugins(): discovered plugin "LDAP Basic" at "/usr/lib/x86_64-linux-gnu/veyon/ldap.so"
2020-10-14T12:12:13.144: [DEBUG] PluginManager::loadPlugins(): discovered plugin "PowerControl" at "/usr/lib/x86_64-linux-gnu/veyon/powercontrol.so"
2020-10-14T12:12:13.145: [DEBUG] PluginManager::loadPlugins(): discovered plugin "RemoteAccess" at "/usr/lib/x86_64-linux-gnu/veyon/remoteaccess.so"
2020-10-14T12:12:13.145: [DEBUG] PluginManager::loadPlugins(): discovered plugin "ScreenLock" at "/usr/lib/x86_64-linux-gnu/veyon/screenlock.so"
2020-10-14T12:12:13.145: [DEBUG] PluginManager::loadPlugins(): discovered plugin "Screenshot" at "/usr/lib/x86_64-linux-gnu/veyon/screenshot.so"
2020-10-14T12:12:13.145: [DEBUG] PluginManager::loadPlugins(): discovered plugin "ServiceControl" at "/usr/lib/x86_64-linux-gnu/veyon/servicecontrol.so"
2020-10-14T12:12:13.145: [DEBUG] PluginManager::loadPlugins(): discovered plugin "Shell" at "/usr/lib/x86_64-linux-gnu/veyon/shell.so"
2020-10-14T12:12:13.145: [DEBUG] PluginManager::loadPlugins(): discovered plugin "SystemUserGroups" at "/usr/lib/x86_64-linux-gnu/veyon/systemusergroups.so"
2020-10-14T12:12:13.146: [DEBUG] PluginManager::loadPlugins(): discovered plugin "TextMessage" at "/usr/lib/x86_64-linux-gnu/veyon/textmessage.so"
2020-10-14T12:12:13.146: [DEBUG] PluginManager::loadPlugins(): discovered plugin "UserSessionControl" at "/usr/lib/x86_64-linux-gnu/veyon/usersessioncontrol.so"
2020-10-14T12:12:13.146: [DEBUG] VeyonCore::initSystemInfo(): "4.4.2" "a13eql11.sanclemente.local" "linux" "5.7.0-0.bpo.2-amd64" "Debian GNU/Linux 10 (buster)" "debian" "10"
2020-10-14T12:12:13.146: [DEBUG] FeatureWorkerManager::startUnmanagedSessionWorker(): Starting worker (unmanaged session process) for feature QUuid("{8e997d84-ebb9-430f-8f72-d45d9821963d}")
2020-10-14T12:12:13.147: [DEBUG] VncProxyServer::start(): started on port 11100
2020-10-14T12:12:13.147: [DEBUG] VncServer::prepare():
2020-10-14T12:12:13.147: [DEBUG] VncServer::run(): running
2020-10-14T12:12:13.149: [DEBUG] VeyonCore::exec(): Running
2020-10-14T12:12:13.213: [DEBUG] FeatureWorkerManager::acceptConnection(): accepting connection
2020-10-14T12:12:13.348: [DEBUG] ServerAuthenticationManager::processAuthenticationMessage(): state VncServerClient::AuthState(Init) type RfbVeyonAuth::Type(KeyFile) host "::ffff:10.0.0.7" user "SANCLEMENTE\\Administrador"
2020-10-14T12:12:13.349: [DEBUG] ServerAuthenticationManager::processAuthenticationMessage(): state VncServerClient::AuthState(Init) type RfbVeyonAuth::Type(KeyFile) host "::ffff:10.0.0.7" user "SANCLEMENTE\\bruno"
2020-10-14T12:12:13.352: [DEBUG] ServerAuthenticationManager::processAuthenticationMessage(): state VncServerClient::AuthState(Challenge) type RfbVeyonAuth::Type(KeyFile) host "::ffff:10.0.0.7" user "SANCLEMENTE\\Administrador"
2020-10-14T12:12:13.352: [DEBUG] ServerAuthenticationManager::performKeyAuthentication(): loading public key "/etc/veyon/keys/public/teacher/key"
2020-10-14T12:12:13.353: [DEBUG] ServerAuthenticationManager::performKeyAuthentication(): SUCCESS
2020-10-14T12:12:13.353: [DEBUG] AccessControlProvider::checkAccess(): no access control method configured, allowing access.
2020-10-14T12:12:13.353: [INFO] ComputerControlServer::showAccessControlMessage(): Access control successful for "::ffff:10.0.0.7" "SANCLEMENTE\\Administrador"
2020-10-14T12:12:13.356: [DEBUG] ServerAuthenticationManager::processAuthenticationMessage(): state VncServerClient::AuthState(Challenge) type RfbVeyonAuth::Type(KeyFile) host "::ffff:10.0.0.7" user "SANCLEMENTE\\bruno"
2020-10-14T12:12:13.356: [DEBUG] ServerAuthenticationManager::performKeyAuthentication(): loading public key "/etc/veyon/keys/public/teacher/key"
2020-10-14T12:12:13.356: [DEBUG] ServerAuthenticationManager::performKeyAuthentication(): SUCCESS
2020-10-14T12:12:13.356: [DEBUG] AccessControlProvider::checkAccess(): no access control method configured, allowing access.
2020-10-14T12:12:13.356: [INFO] ComputerControlServer::showAccessControlMessage(): Access control successful for "::ffff:10.0.0.7" "SANCLEMENTE\\bruno"
2020-10-14T12:12:13.385: [DEBUG] VncClientProtocol::receiveSecurityResult(): authentication successful
2020-10-14T12:12:13.601: [DEBUG] ---
2020-10-14T12:12:13.601: [DEBUG] Last message repeated 1 times
2020-10-14T12:12:13.601: [DEBUG] ---
2020-10-14T12:12:13.601: [DEBUG] AccessControlProvider::checkAccess(): no access control method configured, allowing access.
2020-10-14T12:12:13.601: [INFO] ComputerControlServer::showAccessControlMessage(): Access control successful for "::ffff:10.0.0.7" "SANCLEMENTE\\Administrador"
2020-10-14T12:12:13.605: [DEBUG] VncClientProtocol::receiveSecurityResult(): authentication successful
2020-10-14T12:12:13.606: [DEBUG] ---
2020-10-14T12:12:13.607: [DEBUG] Last message repeated 1 times
2020-10-14T12:12:13.607: [DEBUG] ---
2020-10-14T12:12:13.607: [DEBUG] ServerAuthenticationManager::processAuthenticationMessage(): state VncServerClient::AuthState(Init) type RfbVeyonAuth::Type(KeyFile) host "::ffff:10.0.0.7" user "SANCLEMENTE\\Administrador"
2020-10-14T12:12:13.607: [DEBUG] ServerAuthenticationManager::processAuthenticationMessage(): state VncServerClient::AuthState(Init) type RfbVeyonAuth::Type(KeyFile) host "::ffff:10.0.0.7" user "SANCLEMENTE\\bruno"
2020-10-14T12:12:13.610: [DEBUG] ServerAuthenticationManager::processAuthenticationMessage(): state VncServerClient::AuthState(Challenge) type RfbVeyonAuth::Type(KeyFile) host "::ffff:10.0.0.7" user "SANCLEMENTE\\Administrador"
2020-10-14T12:12:13.610: [DEBUG] ServerAuthenticationManager::performKeyAuthentication(): loading public key "/etc/veyon/keys/public/teacher/key"
2020-10-14T12:12:13.613: [DEBUG] ServerAuthenticationManager::performKeyAuthentication(): SUCCESS
2020-10-14T12:12:13.613: [DEBUG] AccessControlProvider::checkAccess(): no access control method configured, allowing access.
2020-10-14T12:12:13.613: [INFO] ComputerControlServer::showAccessControlMessage(): Access control successful for "::ffff:10.0.0.7" "SANCLEMENTE\\bruno"
2020-10-14T12:12:13.619: [DEBUG] FeatureManager::handleFeatureMessage(): feature QUuid("{79a5e74d-50bd-4aab-8012-0e70dc08cc72}") command 0 arguments QMap()
2020-10-14T12:12:13.619: [DEBUG] ComputerControlServer::sendFeatureMessageReply(): QUuid("{79a5e74d-50bd-4aab-8012-0e70dc08cc72}") 0 QMap(("0", QVariant(QString, ""))("1", QVariant(QString, ""))("2", QVariant(int, -1)))
2020-10-14T12:12:13.619: [DEBUG] FeatureManager::handleFeatureMessage(): feature QUuid("{a0a96fba-425d-414a-aaf4-352b76d7c4f3}") command 0 arguments QMap()
2020-10-14T12:12:13.619: [DEBUG] ComputerControlServer::sendFeatureMessageReply(): QUuid("{a0a96fba-425d-414a-aaf4-352b76d7c4f3}") 0 QMap(("0", QVariant(QStringList, ("{8e997d84-ebb9-430f-8f72-d45d9821963d}"))))
2020-10-14T12:12:13.619: [DEBUG] FeatureManager::handleFeatureMessage(): feature QUuid("{79a5e74d-50bd-4aab-8012-0e70dc08cc72}") command 0 arguments QMap()
2020-10-14T12:12:13.619: [DEBUG] ComputerControlServer::sendFeatureMessageReply(): QUuid("{79a5e74d-50bd-4aab-8012-0e70dc08cc72}") 0 QMap(("0", QVariant(QString, ""))("1", QVariant(QString, ""))("2", QVariant(int, -1)))
2020-10-14T12:12:13.619: [DEBUG] FeatureManager::handleFeatureMessage(): feature QUuid("{79a5e74d-50bd-4aab-8012-0e70dc08cc72}") command 0 arguments QMap()
2020-10-14T12:12:13.619: [DEBUG] ComputerControlServer::sendFeatureMessageReply(): QUuid("{79a5e74d-50bd-4aab-8012-0e70dc08cc72}") 0 QMap(("0", QVariant(QString, ""))("1", QVariant(QString, ""))("2", QVariant(int, -1)))
2020-10-14T12:12:13.620: [DEBUG] FeatureManager::handleFeatureMessage(): feature QUuid("{a0a96fba-425d-414a-aaf4-352b76d7c4f3}") command 0 arguments QMap()
2020-10-14T12:12:13.620: [DEBUG] ComputerControlServer::sendFeatureMessageReply(): QUuid("{a0a96fba-425d-414a-aaf4-352b76d7c4f3}") 0 QMap(("0", QVariant(QStringList, ("{8e997d84-ebb9-430f-8f72-d45d9821963d}"))))
2020-10-14T12:12:13.620: [DEBUG] FeatureManager::handleFeatureMessage(): feature QUuid("{a0a96fba-425d-414a-aaf4-352b76d7c4f3}") command 0 arguments QMap()
2020-10-14T12:12:13.620: [DEBUG] ComputerControlServer::sendFeatureMessageReply(): QUuid("{a0a96fba-425d-414a-aaf4-352b76d7c4f3}") 0 QMap(("0", QVariant(QStringList, ("{8e997d84-ebb9-430f-8f72-d45d9821963d}"))))
2020-10-14T12:12:13.624: [ERR] LinuxSessionFunctions::getSessionProperty(): Could not query session property "Display" "Caller does not belong to any known session"
2020-10-14T12:12:13.627: [ERR] ---
2020-10-14T12:12:13.627: [ERR] Last message repeated 2 times
2020-10-14T12:12:13.627: [ERR] ---
2020-10-14T12:12:13.627: [ERR] LinuxSessionFunctions::getSessionProperty(): Could not query session property "Id" "Caller does not belong to any known session"
2020-10-14T12:12:14.844: [ERR] ---
2020-10-14T12:12:14.844: [ERR] Last message repeated 2 times
2020-10-14T12:12:14.844: [ERR] ---
2020-10-14T12:12:14.844: [DEBUG] FeatureManager::handleFeatureMessage(): feature QUuid("{a0a96fba-425d-414a-aaf4-352b76d7c4f3}") command 0 arguments QMap()
2020-10-14T12:12:14.844: [DEBUG] ComputerControlServer::sendFeatureMessageReply(): QUuid("{a0a96fba-425d-414a-aaf4-352b76d7c4f3}") 0 QMap(("0", QVariant(QStringList, ("{8e997d84-ebb9-430f-8f72-d45d9821963d}"))))
2020-10-14T12:12:14.844: [DEBUG] FeatureManager::handleFeatureMessage(): feature QUuid("{a0a96fba-425d-414a-aaf4-352b76d7c4f3}") command 0 arguments QMap()
2020-10-14T12:12:14.844: [DEBUG] ComputerControlServer::sendFeatureMessageReply(): QUuid("{a0a96fba-425d-414a-aaf4-352b76d7c4f3}") 0 QMap(("0", QVariant(QStringList, ("{8e997d84-ebb9-430f-8f72-d45d9821963d}"))))
2020-10-14T12:12:14.844: [DEBUG] FeatureManager::handleFeatureMessage(): feature QUuid("{79a5e74d-50bd-4aab-8012-0e70dc08cc72}") command 0 arguments QMap()
2020-10-14T12:12:14.844: [DEBUG] ComputerControlServer::sendFeatureMessageReply(): QUuid("{79a5e74d-50bd-4aab-8012-0e70dc08cc72}") 0 QMap(("0", QVariant(QString, "Debian-gdm"))("1", QVariant(QString, ""))("2", QVariant(int, 0)))
2020-10-14T12:12:14.844: [DEBUG] FeatureManager::handleFeatureMessage(): feature QUuid("{79a5e74d-50bd-4aab-8012-0e70dc08cc72}") command 0 arguments QMap()
2020-10-14T12:12:14.844: [DEBUG] ComputerControlServer::sendFeatureMessageReply(): QUuid("{79a5e74d-50bd-4aab-8012-0e70dc08cc72}") 0 QMap(("0", QVariant(QString, "Debian-gdm"))("1", QVariant(QString, ""))("2", QVariant(int, 0)))
2020-10-14T12:12:15.853: [DEBUG] FeatureManager::handleFeatureMessage(): feature QUuid("{a0a96fba-425d-414a-aaf4-352b76d7c4f3}") command 0 arguments QMap()
2020-10-14T12:12:15.853: [DEBUG] ComputerControlServer::sendFeatureMessageReply(): QUuid("{a0a96fba-425d-414a-aaf4-352b76d7c4f3}") 0 QMap(("0", QVariant(QStringList, ("{8e997d84-ebb9-430f-8f72-d45d9821963d}"))))

```

![Captura](https://user-images.githubusercontent.com/6134961/95967162-3a926080-0e0c-11eb-8db9-5d3b73d9b44f.PNG)

**Comment by tobydox** [comment]
Can you please try to update to 4.5.0 and report if the issue still persists?

**Comment by tobydox** [comment]
Issue fixed in latest 4.5.4 release.
